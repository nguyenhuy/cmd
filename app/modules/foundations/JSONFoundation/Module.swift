Target.module(
  name: "JSONFoundation",
  dependencies: [],
  testDependencies: [])
