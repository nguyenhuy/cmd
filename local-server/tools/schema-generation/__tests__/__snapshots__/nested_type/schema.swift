// swiftformat:disable all
// This file is generated from ./local-server/src/server/schemas/schemaFile.ts by `yarn export-schema-swift`.
// Do not edit it manually.

import Foundation
import JSONFoundation

extension Schema {
  public struct ValueType: Codable, Sendable {
    public let nestedValue: NestedValueType
  
    private enum CodingKeys: String, CodingKey {
      case nestedValue = "nested_value"
    }
  
    public init(
        nestedValue: NestedValueType
    ) {
      self.nestedValue = nestedValue
    }
  
    public init(from decoder: Decoder) throws {
      let container = try decoder.container(keyedBy: CodingKeys.self)
      nestedValue = try container.decode(NestedValueType.self, forKey: .nestedValue)
    }
  
    public func encode(to encoder: Encoder) throws {
      var container = encoder.container(keyedBy: CodingKeys.self)
      try container.encode(nestedValue, forKey: .nestedValue)
    }
  }
  public struct NestedValueType: Codable, Sendable {
    public let value: String
  
    private enum CodingKeys: String, CodingKey {
      case value = "value"
    }
  
    public init(
        value: String
    ) {
      self.value = value
    }
  
    public init(from decoder: Decoder) throws {
      let container = try decoder.container(keyedBy: CodingKeys.self)
      value = try container.decode(String.self, forKey: .value)
    }
  
    public func encode(to encoder: Encoder) throws {
      var container = encoder.container(keyedBy: CodingKeys.self)
      try container.encode(value, forKey: .value)
    }
  }}
