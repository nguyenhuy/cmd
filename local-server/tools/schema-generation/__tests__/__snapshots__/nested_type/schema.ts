export interface ValueType {
  nested_value: NestedValueType;
}
export interface NestedValueType {
  value: string;
}
