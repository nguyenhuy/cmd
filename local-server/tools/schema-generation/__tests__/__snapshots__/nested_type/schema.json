{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "ValueType": {
      "type": "object",
      "properties": {
        "nested_value": {
          "$ref": "#/definitions/NestedValueType"
        }
      },
      "required": [
        "nested_value"
      ],
      "additionalProperties": false
    },
    "NestedValueType": {
      "type": "object",
      "properties": {
        "value": {
          "type": "string"
        }
      },
      "required": [
        "value"
      ],
      "additionalProperties": false
    }
  }
}